src/Login/index.js

```diff
@@ -0,0 +1,59 @@
+import React, {Component, PropTypes} from 'react';
+import {themr} from 'react-css-themr';
+import {connect} from 'react-redux';
+
+import {setUsername, setPassword, toggleRemember, attemptLogin} from './actions';
+import style from './style.scss';
+import constants from '../constants';
+
+class Login extends Component {
+    static propTypes = {
+        theme: PropTypes.object,
+        username: PropTypes.string,
+        password: PropTypes.string,
+        remember: PropTypes.bool,
+        onUsernameChange: PropTypes.func,
+        onPasswordChange: PropTypes.func,
+        onRememberToggle: PropTypes.func,
+        onLogin: PropTypes.func
+    }
+
+    render() {
+        const {theme, username, password, remember, onUsernameChange, onPasswordChange, onRememberToggle, onLogin} = this.props;
+        return (
+            <div className={theme.container}>
+                <div className={theme.property}>
+                    <label className={theme.propertyLabel} htmlFor='uname'>{constants.USERNAME}</label>
+                    <input type='text' id='uname' placeholder={constants.USERNAME} required={true} value={username} onChange={(e) => { onUsernameChange(e.target.value); }}/>
```

Comment:
I believe that is still the job of the component rather than the connecting logic. The connector should not be responsible for knowing that a button is going to cause a page transition if its default behavior isn't cancelled. The component should know that.